### PR TITLE
add `dolt_ci_workflows` table for storing DoltHub/DoltLab continuous integration configuration

### PR DESCRIPTION
**Pull request by coffeegoddd**

### CODE REVIEW

**Comment by coffeegoddd** [comment]
@coffeegoddd __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>f9b96f3</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>f9b96f3</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

File: go/libraries/doltcore/doltdb/commit_hooks_test.go [hunk 1]
```diff
@@ -104,7 +105,16 @@ func TestPushOnWriteHook(t *testing.T) {
 
 	names, err := root.GetTableNames(context.Background(), DefaultSchemaName)
 	assert.NoError(t, err)
-	if len(names) != 0 {
+
+	// ignore dolt_ci_* tables
+	filtered := make([]string, 0)
```
Comment (zachmu) [inline go/libraries/doltcore/doltdb/commit_hooks_test.go:110]: Seems like there shouldn't be any dolt_ci tables here if you've never inserted into any

File: go/libraries/doltcore/doltdb/doltdb.go [hunk 2]
```diff
@@ -209,6 +213,11 @@ func (ddb *DoltDB) WriteEmptyRepoWithCommitMetaGeneratorAndDefaultBranch(
 		return err
 	}
 
+	rv, err = ddb.ciTablesCreator.CreateCIConfigTables(ctx, rv)
```
Comment (zachmu) [inline go/libraries/doltcore/doltdb/doltdb.go:216]: You don't want to do this. The pattern you're using in `database.go`, which is correct, is that this table always implicitly exists, but never gets written to the root value unless you put data in it..

**Review by zachmu — COMMENTED** [review]
On the right track. Main issue is explicitly creating these tables at database init -- you don't want to do that. See comments.

**Comment by coffeegoddd** [comment]
@zachmu @Hydrocharged so If i do not create the tables at the time of `dolt init`, then I get a panic whenever I try to query the `dolt_ci_workflows` table. 

If I put in an empty `DoltTable`, like so https://github.com/dolthub/dolt/pull/8257/files#diff-d90afb53621f16e22592fede5474999954c2042529f03d424701bd3f1746237bR113 , it resolves the panic, but then `e.Analyzer.ExecBuilder.Build` from engine.go in gms returns `table not found: dolt_ci_workflows`.

Additionally, the next table I plan to add has a foreign key reference to `dolt_ci_workflows` and i'm not sure how that will behave if this table is ephemeral.

**Comment by coffeegoddd** [comment]
@coffeegoddd __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>b43ea57</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>b43ea57</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Pull request closed without merging**